internal/services/mssql/mssql_database_resource.go

```diff
@@ -488,133 +492,175 @@ func resourceMsSqlDatabaseRead(d *pluginsdk.ResourceData, meta interface{}) erro
-		}
-		d.Set("min_capacity", props.MinCapacity)
-		d.Set("read_replica_count", props.HighAvailabilityReplicaCount)
-		if props.ReadScale == sql.DatabaseReadScaleEnabled {
-			d.Set("read_scale", true)
-		} else if props.ReadScale == sql.DatabaseReadScaleDisabled {
-			d.Set("read_scale", false)
-		}
-		if props.CurrentServiceObjectiveName != nil {
-			skuName = *props.CurrentServiceObjectiveName
-		}
-		d.Set("sku_name", skuName)
-		d.Set("storage_account_type", string(props.CurrentBackupStorageRedundancy))
-		d.Set("zone_redundant", props.ZoneRedundant)
-		if props.IsLedgerOn != nil {
-			ledgerEnabled = *props.IsLedgerOn
-		}
 
-		configurationName := ""
-		if v := props.MaintenanceConfigurationID; v != nil {
-			maintenanceConfigId, err := publicmaintenanceconfigurations.ParsePublicMaintenanceConfigurationIDInsensitively(*v)
-			if err != nil {
-				return err
+	if model := resp.Model; model != nil {
+		d.Set("name", id.DatabaseName)
+
+		if props := model.Properties; props != nil {
+			if props.AutoPauseDelay != nil {
+				d.Set("auto_pause_delay_in_minutes", props.AutoPauseDelay)
 			}
-			configurationName = maintenanceConfigId.PublicMaintenanceConfigurationName
-		}
-		d.Set("maintenance_configuration_name", configurationName)
 
-		d.Set("ledger_enabled", ledgerEnabled)
-	}
+			if props.Collation != nil {
+				d.Set("collation", props.Collation)
+			}
 
-	securityAlertPolicy, err := securityAlertPoliciesClient.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
-	if err == nil {
-		if err := d.Set("threat_detection_policy", flattenMsSqlServerSecurityAlertPolicy(d, securityAlertPolicy)); err != nil {
-			return fmt.Errorf("setting `threat_detection_policy`: %+v", err)
-		}
-	}
+			if props.ElasticPoolId != nil {
+				elasticPoolId = pointer.From(props.ElasticPoolId)
+			}
 
-	geoBackupPolicy := true
+			if props.MinCapacity != nil {
+				minCapacity = pointer.From(props.MinCapacity)
+			}
 
-	// DW SKU's do not currently support LRP and do not honour normal SRP operations
-	if !strings.HasPrefix(skuName, "DW") {
-		longTermPolicy, err := longTermRetentionClient.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
-		if err != nil {
-			return fmt.Errorf("retrieving Long Term Retention Policies for %s: %+v", id, err)
-		}
-		if err := d.Set("long_term_retention_policy", helper.FlattenLongTermRetentionPolicy(&longTermPolicy, d)); err != nil {
-			return fmt.Errorf("setting `long_term_retention_policy`: %+v", err)
-		}
+			if props.HighAvailabilityReplicaCount != nil {
+				d.Set("read_replica_count", props.HighAvailabilityReplicaCount)
+			}
 
-		shortTermPolicy, err := shortTermRetentionClient.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
-		if err != nil {
-			return fmt.Errorf("retrieving Short Term Retention Policies for %s: %+v", id, err)
-		}
+			if props.RequestedBackupStorageRedundancy != nil {
+				d.Set("storage_account_type", string(pointer.From(props.RequestedBackupStorageRedundancy)))
+			}
```

Comment:
this always wants a value set, so can we make this one:

```suggestion
			requestedBackupStorageRedundancy := ""
			if props.RequestedBackupStorageRedundancy != nil {
				requestedBackupStorageRedundancy = string(*props.RequestedBackupStorageRedundancy))
			}
			d.Set("storage_account_type", requestedBackupStorageRedundancy)
```

Reply:
Fixed.

Reply:
Looks like this one is still pending?

Reply:
No, this was changed but moved up a few lines(line 527 I believe now), I just forgot to remove this legacy code so it looked like I didn't actually change the code... my bad, sorry.